adc/files.go

```diff
@@ -19,17 +25,85 @@ func (GetInfoRequest) Cmd() MsgType {
 	return MsgType{'G', 'F', 'I'}
 }
 
+var (
+	_ Marshaler   = GetRequest{}
+	_ Unmarshaler = (*GetRequest)(nil)
+)
+
 type GetRequest struct {
-	Type  string `adc:"#"`
-	Path  string `adc:"#"`
-	Start int64  `adc:"#"`
-	Bytes int64  `adc:"#"`
+	Type       string
+	Path       string
+	Start      int64
+	Bytes      int64
+	Compressed bool
 }
 
 func (GetRequest) Cmd() MsgType {
 	return MsgType{'G', 'E', 'T'}
 }
 
+func (m GetRequest) MarshalADC(buf *bytes.Buffer) error {
+	buf.Write(escape(m.Type))
+	buf.WriteByte(' ')
+	buf.Write(escape(m.Path))
+	buf.WriteByte(' ')
+	buf.WriteString(strconv.FormatInt(m.Start, 10))
+	buf.WriteByte(' ')
+	buf.WriteString(strconv.FormatInt(m.Bytes, 10))
+	if m.Compressed {
+		buf.Write([]byte(" ZL1"))
+	}
+	return nil
+}
+
+func (m *GetRequest) UnmarshalADC(data []byte) error {
+	i := bytes.IndexByte(data, ' ')
```

Comment:
Thanks a lot for taking the time to reimplement it!

I think it's still possible to use the generic reflection code for this, but the manually written code will be much faster than the reflection.